tests/driver/api.test.ts

```diff
@@ -807,6 +807,19 @@ describe('Mongoose Model API level tests', async () => {
             assert.deepStrictEqual($vector, [1, 101]);
         });
 
+        it('supports sort() and similarity score with $meta with find()', async function() {
+            const res = await Vector.find({}, { name: 1, $similarity : 1}).sort({ $vector: { $meta: [1, 99] } });
+            assert.deepStrictEqual(res.map(doc => doc.name), ['Test vector 1', 'Test vector 2']);
+            assert.deepStrictEqual(res.map(doc => doc.get('$similarity')), [1, 0.51004946]);
```

Comment:
This `doc.get()` is a bit clunky, but you're right that's the only way you can access a property that's returned from the database but not in your schema.

One potential alternative is to use a virtual for `$similarity` as follows:

```javascript
  schema.virtual('$similarity').
    get(function() { return this._doc.$similarity; });
```

That will allow you to do `doc.$similarity` without needing to use `get()`.

Reply:
I was not aware, we could use `virtual` for this. Good to know, thanks!